tsdb/wal/watcher.go

```diff
@@ -483,6 +488,10 @@ func (w *Watcher) readSegment(r *LiveReader, segmentNum int, tail bool) error {
 			}
 			for _, s := range samples {
 				if s.T > w.StartTime {
+					if !sentSamples && segmentNum == w.lastSegment {
```

Comment:
Why the check for `segmentNum == w.lastSegment` here? I would have thought that would be controlled by `tail`?

Reply:
`w.lastSegment` is the last/newest segment that existed on startup. We only want to log `done replaying WAL` once, when we find a sample with a timestamp `s.T > w.StartTime` in the newest segment that existed when the watcher started. It's possible that new segments were created during the replay, but even if there weren't we'd still end up logging this for every new segment that's created while tailing the WAL.

Reply:
I think it would be simpler to store `sentSamples` rather than `lastSegment` on the `Watcher` struct. Then there is only one check that is more obvious. WDYT?

Otherwise, I think `lastSegment` needs to be named something else as I would expect a variable with that name to change over time as more segments are created.

Reply:
I changed it to just `sendSamples`, which is set to true when we find a sample after the `StartTime`.